source/Cosmos.System2/Console.cs

```diff
@@ -239,6 +367,84 @@ public int CursorSize
             }
         }
 
+        public Stream OpenStandardInput()
+        {
+            return new CosmosConsoleStream(FileAccess.Read);

```

Comment:
can we use the same instance and not make a new one each time this is called? same for the other 2 methods

Reply:
Could be make, since Error and oytput are a stream with only write access, should they both be the same stream or different ones?

Reply:
i was thinking of something like this, unless we want a new stream for each time its called
```cs
Stream? stdOut;
public Stream OpenStandardInput()
{
    if(stdOut == null)
        stdOut = new CosmosConsoleStream(FileAccess.Read);
    return stdOut;
}
```

Reply:
Can also be:
```csharp
Stream? stdOut;
public Stream OpenStandardInput()
{
    return stdOut ?? = new CosmosConsoleStream(FileAccess.Read);
}
```

it really does not matter to much if it is the same string or a completly new one, if you see the code of the `Cosmos.Console.cs` you will see that all it does is just point directly to de Cosmos.System.Console methods. Different instances would be a case if we want it to work as a normal Stream(Do not really write content till you call the Flush method), but since the content is not saved on a buffer it really does not matter.

Reply:
o if it does not matter then i would say reuse the same one better for GC

Reply:
Will do

Reply:
> Could be make, since Error and output are a stream with only write access, should they both be the same stream or different ones?

Wich one do you think is a better choice? the same one to have less objects?

Reply:
i think not as they are not the same so 2 instances

Reply:
Done

Reply:
Now i will look into why the `Console.WriteLine(format, string)` is not working. I think it have to do something with how StreamWriter get's a StringBuilder object, but i do not know. maybe there is a method to plug.